test/TestProjects/Spread-Cadl/Generated/Models/SpreadAliasWithOptionalPropsRequest.Serialization.cs

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+using System.Text.Json;
+using Azure.Core;
+
+namespace Spread.Models
+{
+    internal partial class SpreadAliasWithOptionalPropsRequest : IUtf8JsonSerializable
+    {
+        void IUtf8JsonSerializable.Write(Utf8JsonWriter writer)
+        {
+            writer.WriteStartObject();
+            writer.WritePropertyName("name");
+            writer.WriteStringValue(Name);
+            if (Optional.IsDefined(Color))
+            {
+                writer.WritePropertyName("color");
+                writer.WriteStringValue(Color);
+            }
+            if (Optional.IsDefined(Age))
+            {
+                if (Age != null)
+                {
+                    writer.WritePropertyName("age");
+                    writer.WriteNumberValue(Age.Value);
+                }
+                else
+                {
+                    writer.WriteNull("age");
+                }
+            }
```

Comment:
Both color and age are optional, why they have different behaviors? i.e., why there is no null check insider Color block.

Reply:
This is not something I introduced in this PR.